src/arch/riscv/irqc/aia/inc/vaplic.h

```diff
@@ -0,0 +1,82 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Bao Project and Contributors. All rights reserved.
+ */
+
+#ifndef __VAPLIC_H__
+#define __VAPLIC_H__
+
+#include <bao.h>
+#include <aplic.h>
+#include <arch/spinlock.h>
+#include <bitmap.h>
+#include <emul.h>
+
+/**
+ * @brief   Data structure to the virtual interrupt controller when a APLIC 
+ *          is the platform interrupt controller.
+ * 
+ */
+struct virqc {
+    spinlock_t lock;
+    size_t idc_num;
+    BITMAP_ALLOC(hw, APLIC_MAX_INTERRUPTS);
+    uint32_t domaincfg;
+    uint32_t srccfg[APLIC_MAX_INTERRUPTS-1];
+    uint32_t setip[APLIC_MAX_INTERRUPTS/32];
+    uint32_t in_clrip[APLIC_MAX_INTERRUPTS/32];
```

Comment:
I added this register to the struct because I wasn't sure how to virtualise it.
Actually, I think it's simpler than I thought. This register should return the rectified interrupt. Therefore, we just need to read this register from the APLIC, masking it with the interrupts connected to the hardware. Virtually we can't touch an interrupt line (only the pending and enable state), right?... So those will always be 0. 
 What is your opinion on this?

Reply:
This variable disappear since the vaplic_get_inclrip function is implemented as follow:

```
static uint32_t vaplic_get_in_clrip(struct vcpu *vcpu, uint8_t reg){
    struct vaplic *vaplic = &vcpu->vm->arch.vaplic;
    uint32_t ret = 0;
    if (reg < APLIC_NUM_CLRIx_REGS) ret = (aplic_get_inclrip(reg) & 
                                           vaplic->hw[reg]);
    return ret;
}
```